### PR TITLE
submodules now properly submoduled

### PR DESCRIPTION
**Pull request by sbromberger**

### CODE REVIEW

**Comment by sbromberger** [comment]
@jpfairbanks - this incorporates `spectral.jl` into `LightGraphs.LinAlg` and makes the latter a real submodule. I've also refactored the tests to test the submodules appropriately.

File: src/linalg/LinAlg.jl [hunk 1]
```diff
@@ -1,6 +1,54 @@
 module LinAlg
-include("./graphmatrices.jl")
-end
 
+using SimpleTraits
+using ..LightGraphs
+
+import LightGraphs: IsDirected, adjacency_matrix, laplacian_matrix, laplacian_spectrum
```
Comment (iamed2) [inline src/linalg/LinAlg.jl]: Use `.LightGraphs` here to refer to the module you just imported with `using ..LightGraphs`

File: src/linalg/LinAlg.jl [hunk 2]
```diff
@@ -1,6 +1,54 @@
...
+	StochasticLaplacian,
+	AveragingAdjacency,
+	AveragingLaplacian,
+	PunchedAdjacency,
+	Noop,
+	diag,
+	degrees,
+	symmetrize,
+	prescalefactor,
+    postscalefactor,
```
Comment (iamed2) [inline src/linalg/LinAlg.jl]: The following lines and previous lines are indented differently

File: test/linalg/runtests.jl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# might error out on very small eltypes.
+testlargegraphs(g) = [g, Graph{UInt16}(g), Graph{Int32}(g)]
+testlargedigraphs(g) = [g, DiGraph{UInt16}(g), DiGraph{Int32}(g)]
+
+tests = [
+    "graphmatrices",
+    "spectral"
+]
+
+@testset "LightGraphs.LinAlg" begin
```
Comment (iamed2) [inline test/linalg/runtests.jl]: I haven't seen testsets in a module before. Is there an example of this in Base or in the wild?

I would make a function in `TestLinAlg` called `test_linalg()` or something and put this block in there, then call that function from outside the module in this file. Does that make sense?

Same think with the SimpleGraph tests.

File: src/linalg/LinAlg.jl (hunk 2 above)
Comment (sbromberger) [reply src/linalg/LinAlg.jl]: Thanks. This was because someone was using tabs instead of spaces. :)

File: test/linalg/runtests.jl (hunk 3 above)
Comment (sbromberger) [reply test/linalg/runtests.jl]: I haven't either, actually. I'll expose a `test_linalg()`.

File: src/linalg/LinAlg.jl (hunk 1 above)
Comment (sbromberger) [reply src/linalg/LinAlg.jl]: That resulted in `ERROR: LoadError: LoadError: in module path: LightGraphs not defined`

Comment (iamed2) [reply src/linalg/LinAlg.jl]: You did `import .LightGraphs: IsDirected, ...`?

:/ I guess never mind. It's weird to me that that didn't work.

File: test/linalg/runtests.jl (hunk 3 above)
Comment (sbromberger) [reply test/linalg/runtests.jl]: Actually, I'm just collapsing this by getting rid of the module entirely.

File: src/linalg/LinAlg.jl (hunk 1 above)
Comment (sbromberger) [reply src/linalg/LinAlg.jl]: > You did import .LightGraphs: IsDirected, ...?

Precisely that.

**Comment by sbromberger** [comment]
Thank you, @jpfairbanks!

**Pull request merged**